docs/user/trusted-publishers/internals.md

```diff
@@ -176,8 +179,31 @@ used by the Python packaging community:
    can be published from both `release-linux.yml` and `release-macos.yml`
    without needing to be refactored into a single `release.yml`.
 
+### What are account resurrection attacks, and how does PyPI protect against them?
+
+Some OIDC providers support username changes, so a claim of
+`repository_owner: octo-org` might not necessarily refer to the same `octo-org`
+that a user initially authorized in a trusted publisher configuration.
+
+If a repository owner changes their username or deletes their account, a
+malicious actor can take the freed username and create their own repositories
+under the original trusted name. This is known as an *account resurrection
+attack*.
+
+To solve this issue, PyPI worked with GitHub to add the `repository_owner_id`
+claim to OIDC tokens. This claim attests to the ID of the repository owner,
+which is stable and permanent unlike usernames. When a trusted publisher is
+configured, PyPI looks up the configured username's ID and stores it. During
+API token minting, PyPI checks the `repository_owner_id` claim against the
+stored ID and fails if they don't match. Through this process, only the original
+GitHub user remains authorized to publish to their PyPI projects, even if they
+change their username or delete their account.
```

Comment:
We probably don't need the specific history here 🙂 -- we can just say "for GitHub-based publishers, PyPI always checks the `repository_owner_id` claim, since ..."